Binary files hacktv/.git/index and hacktv_patched/.git/index differ
diff '-x=.git' -uNr hacktv/.git/logs/HEAD hacktv_patched/.git/logs/HEAD
diff '-x=.git' -uNr hacktv/.git/logs/HEAD.orig hacktv_patched/.git/logs/HEAD.orig
diff '-x=.git' -uNr hacktv/.git/logs/HEAD.rej hacktv_patched/.git/logs/HEAD.rej
diff '-x=.git' -uNr hacktv/.git/logs/refs/heads/master hacktv_patched/.git/logs/refs/heads/master
diff '-x=.git' -uNr hacktv/.git/logs/refs/heads/master.orig hacktv_patched/.git/logs/refs/heads/master.orig
diff '-x=.git' -uNr hacktv/.git/logs/refs/heads/master.rej hacktv_patched/.git/logs/refs/heads/master.rej
diff '-x=.git' -uNr hacktv/.git/logs/refs/remotes/origin/HEAD hacktv_patched/.git/logs/refs/remotes/origin/HEAD
diff '-x=.git' -uNr hacktv/.git/logs/refs/remotes/origin/HEAD.orig hacktv_patched/.git/logs/refs/remotes/origin/HEAD.orig
diff '-x=.git' -uNr hacktv/.git/logs/refs/remotes/origin/HEAD.rej hacktv_patched/.git/logs/refs/remotes/origin/HEAD.rej
diff '-x=.git' -uNr hacktv/Makefile hacktv_patched/Makefile
--- hacktv/Makefile	2020-07-13 13:56:09.723050400 +0200
+++ hacktv_patched/Makefile	2020-07-13 13:25:03.798978200 +0200
@@ -3,7 +3,7 @@
 PKGCONF := $(CROSS_HOST)pkg-config
 CFLAGS  := -g -Wall -Wno-unused-result -pthread -O3 $(EXTRA_CFLAGS)
 LDFLAGS := -g -lm -pthread $(EXTRA_LDFLAGS)
-OBJS    := hacktv.o common.o fir.o vbidata.o teletext.o wss.o video.o mac.o videocrypt.o videocrypts.o syster.o systerdes.o acp.o nicam728.o test.o ffmpeg.o file.o hackrf.o font.o subtitles.o bitmap.o eurocrypt.o
+OBJS    := hacktv.o common.o keyboard.o fir.o vbidata.o teletext.o wss.o video.o mac.o videocrypt.o videocrypts.o syster.o systerdes.o acp.o nicam728.o test.o ffmpeg.o file.o hackrf.o font.o subtitles.o bitmap.o eurocrypt.o
 PKGS    := libavcodec libavformat libavdevice libswscale libswresample libavutil libhackrf libavfilter freetype2 $(EXTRA_PKGS)
 
 SOAPYSDR := $(shell $(PKGCONF) --exists SoapySDR && echo SoapySDR)
diff '-x=.git' -uNr hacktv/cards/funcard/Videocrypt-All-Modes-ATMEL.eep hacktv_patched/cards/funcard/Videocrypt-All-Modes-ATMEL.eep
--- hacktv/cards/funcard/Videocrypt-All-Modes-ATMEL.eep	2020-07-13 13:56:09.731029100 +0200
+++ hacktv_patched/cards/funcard/Videocrypt-All-Modes-ATMEL.eep	1970-01-01 01:00:00.000000000 +0100
@@ -1,5 +0,0 @@
-:100000000100112233445566778899AABBCCDDEEF6
-:10001000FF489B4DA6F9D9DF6EAC84FA8B2EB676DD
-:1000200019C1B0A3BB0CFD7072CA55EFA07FBF59B8
-:01003000AD22
-:00000001FF
diff '-x=.git' -uNr hacktv/cards/funcard/Videocrypt-All-Modes-ATMEL.hex hacktv_patched/cards/funcard/Videocrypt-All-Modes-ATMEL.hex
--- hacktv/cards/funcard/Videocrypt-All-Modes-ATMEL.hex	2020-07-13 13:56:09.732027600 +0200
+++ hacktv_patched/cards/funcard/Videocrypt-All-Modes-ATMEL.hex	1970-01-01 01:00:00.000000000 +0100
@@ -1,283 +0,0 @@
-:100000000CC0189518951895189518951895189569
-:10001000189518951895189518958FE58DBFC8E314
-:10002000E0E24E2E82E08EBFD2E0F2E05F2EEEEFF5
-:10003000F1E0A0E6B0E088278D933197E9F76624D8
-:1000400080E895E2A6E6B0E08D939C9381E08093F2
-:100050006C0080E080936500009165000F3010F027
-:1000600009F01AC0ACE6B0E0EC91FF27A5E6B0E0DD
-:10007000AD90BB24AFE6B0E0AA0DBB1D91E0A2D7C6
-:10008000ACE6B0E08C918F5F8C93A5E6B0E08C91EC
-:100090008F5F8C9308F4E0CF80E08093650000913F
-:1000A00065000F3110F009F01AC0ACE6B0E0EC9139
-:1000B000FF27A5E6B0E0AD90BB24ABEAB0E0AA0D07
-:1000C000BB1D91E07FD7ACE6B0E08C918F5F8C9345
-:1000D000A5E6B0E08C918F5F8C9308F4E0CFA4D1BB
-:1000E00080E080936C0080E080936E00EAE7FDE0A2
-:1000F00086D68FE0A42DB52D8C9381E00AD7C8D584
-:10010000229681E002D781E080936E0080E08093A8
-:100110006A00A6E6B0E08D919C91ABEDB0E08D93C6
-:100120009C93DDD607D881E08A9382E08A9388E0A9
-:100130008A93A0E6B0E085E08A93E7E386E0E8945E
-:100140009ED700916100013D09F07BC081ED809355
-:100150006800A6E6B0E08D919C91ABEDB0E08D9388
-:100160009C93BDD6E7D781E08A9382E08A9388E0AA
-:100170008A9381E08A93A8E6B0E0E7E386E0E8941A
-:1001800028D780E080936500009165000F3110F062
-:1001900009F031C0A6E6B0E08D919C91ABEDB0E0E6
-:1001A0008D939C939CD6C6D781E08A9382E08A93F4
-:1001B00088E08A93A5E6B0E0AD90BB24A3E8B0E068
-:1001C000AA0DBB1D81E08A93E7E386E0E89457D748
-:1001D000A5E6B0E0AD90BB24A3E8B0E0AA0DBB1D3E
-:1001E000BA93AA938FD02296A5E6B0E08C918F5F48
-:1001F0008C9308F4C9CF00918300083F09F004C034
-:1002000082E080936D000FC000918900003209F0F8
-:1002100004C080E080936D0006C0033609F003C07F
-:1002200081E080936D0000916D00003009F001C005
-:1002300006D300916D00013009F001C03ED166D5B2
-:100240005EC0013E09F021C000916D00023008F04F
-:1002500002C096D419C000918800053809F004C086
-:10026000ECE9FDE0CCD506C0013D09F003C0E6EAAB
-:10027000FDE0C5D588E0A42DB52D8C9381E049D64D
-:1002800007D5229681E041D63AC0013C09F00EC064
-:10029000EAE8FDE0B4D580E1A42DB52D8C9381E092
-:1002A00038D6F6D4229681E030D629C0013A09F03A
-:1002B00025C000916C00053008F009C0E0EBFDE0BE
-:1002C0009ED5ACE6B0E08C918F5F8C930CC0009112
-:1002D0006B00003009F404C0E4EEFDE090D503C0EB
-:1002E000EAECFDE08CD589E1A42DB52D8C9381E05D
-:1002F00010D6CED4229681E008D601C007D509CF0A
-:10030000F894FFCFA881B9810C914AEA0423AAE6A8
-:10031000B0E00C9391E0AAE6B0E014D6A881B981D0
-:100320008C9185758C9391E0A881B981F8D5009165
-:100330006A00A881B9814C91042BA881B9810C93E2
-:10034000A881B9810C914CEC0423AAE6B0E00C938F
-:1003500092E0AAE6B0E0F6D5A881B9818C918373CA
-:100360008C9392E0A881B981DAD500916A00A881C6
-:10037000B9814C91042BA881B9810C93A881B981D2
-:100380008C9182958C93A881B9818C918C939FEFED
-:1003900089278C93089500918E008091910080931D
-:1003A0008E00A1E9B0E00C9300918F008091900045
-:1003B00080938F00A0E9B0E00C93009192008091AF
-:1003C000950080939200A5E9B0E00C930091930012
-:1003D0008091940080939300A4E9B0E00C93009185
-:1003E00096008091990080939600A9E9B0E00C9363
-:1003F000009197008091980080939700A8E9B0E061
-:100400000C9300919A0080919D0080939A00ADE931
-:10041000B0E00C9300919B0080919C0080939B0026
-:10042000ACE9B0E00C93089500916F0080917200E8
-:1004300080936F00A2E7B0E00C9300917000809170
-:10044000710080937000A1E7B0E00C9300917300FD
-:100450008091760080937300A6E7B0E00C93009142
-:1004600074008091750080937400A5E7B0E00C9350
-:100470000091770080917A0080937700AAE7B0E03E
-:100480000C93009178008091790080937800A9E71F
-:10049000B0E00C9300917B0080917E0080937B0004
-:1004A000AEE7B0E00C9300917C0080917D008093DA
-:1004B0007C00ADE7B0E00C93089580E080936B0082
-:1004C0006ADF00E010E020E030E0ABECB0E00D933C
-:1004D0001D932D933C9309EB19E727E33EE9AFEC1D
-:1004E000B0E00D931D932D933C9380E080936500C5
-:1004F000009165000F3110F009F07DC1A2E9B0E074
-:100500000D911D912D913C91A3EDB0E00D931D93A4
-:100510002D933C9394E0A3EDB0E009D5A2E9B0E0BF
-:100520000D911D912D913C91A7EDB0E00D931D9380
-:100530002D933C9395E0A7EDB0E00CD5A3EDB0E092
-:100540000D911D912D913C91A7EDB0E04D915D91E4
-:100550006D917C91B1D4A3EDB0E00D931D932D93DB
-:100560003D93A3EDB0E00D911D912D913C91A2E939
-:10057000B0E04D915D916D917C9111D5A3EDB0E00E
-:100580000D931D932D933D93ABECB0E00D911D9118
-:100590002D913C9143E050E060E070E092D4A7EDF3
-:1005A000B0E00D931D932D933D93ABECB0E00D9116
-:1005B0001D912D913C91A7EDB0E0AD90BD90AFE6BF
-:1005C000B0E0AA0CBB1CAA0CBB1CAA0DBB1D4D9114
-:1005D0005D916D917C91E3D4A7EDB0E00D931D93F7
-:1005E0002D933D93A3EDB0E00D911D912D913C9184
-:1005F000A7EDB0E04D915D916D917C915DD4A3ED3F
-:10060000B0E00D931D932D933D93AEE8B0E00D91B6
-:100610001D912D913C91A3EDB0E04D915D916D91B7
-:100620007C91BDD4AEE8B0E00D931D932D933D9326
-:10063000ABECB0E00D911D912D913C91AFECB0E091
-:100640004D915D916D917C91AAD4ABECB0E00D938E
-:100650001D932D933D93AEE8B0E00D911D912D912A
-:100660003C91A3EDB0E00D931D932D933C9394E04A
-:10067000A3EDB0E05CD4AEE8B0E00D911D912D91FA
-:100680003C91A7EDB0E00D931D932D933C9395E025
-:10069000A7EDB0E05FD4A3EDB0E00D911D912D91D9
-:1006A0003C91A7EDB0E04D915D916D917C9104D4AA
-:1006B000A3EDB0E00D931D932D933D93AEE8B0E014
-:1006C0000D911D912D913C91A3EDB0E04D915D9167
-:1006D0006D917C9164D4A3EDB0E00D931D932D93A7
-:1006E0003D93ABECB0E00D911D912D913C91A7EDA8
-:1006F000B0E00D931D932D933C939BE0A7EDB0E0EC
-:1007000029D4A7EDB0E00D911D912D913C9143E0CE
-:1007100050E060E070E0D5D3A7EDB0E00D931D93FD
-:100720002D933D93ABECB0E00D911D912D913C913B
-:10073000A7EDB0E0AD90BD90AFE6B0E0AA0CBB1C59
-:10074000AA0CBB1CAA0DBB1D4D915D916D917C91B6
-:1007500026D4A7EDB0E00D931D932D933D93A3ED0B
-:10076000B0E00D911D912D913C91A7EDB0E04D9120
-:100770005D916D917C91A0D3A3EDB0E00D931D939D
-:100780002D933D93A3EDB0E00D911D912D913C91E2
-:10079000A2E9B0E04D915D916D917C9100D4A2E908
-:1007A000B0E00D931D932D933D93009165000730AC
-:1007B00009F01AC0AEE8B0E04D915D916D917C9169
-:1007C000A6E9B0E00D911D912D913C9104171507FC
-:1007D0002607370709F408C0ABE6B0E08C918160CA
-:1007E0008C938FEF80936C00A5E6B0E08C918F5FC7
-:1007F0008C9308F47DCECFDD80E08093650000917E
-:100800006500073010F009F019C0A5E6B0E0AD9022
-:10081000BB24AEE8B0E0AA0DBB1D8C91A5E6B0E00C
-:10082000AD90BB24A3EAB0E0AA0DBB1D8C93A5E656
-:10083000B0E08C918F5F8C9308F4E1CF089580E055
-:1008400080936B0080E080937F0080E08093800045
-:1008500080E08093650000916500073010F009F09A
-:1008600011C080E0A5E6B0E0AD90BB24A3EAB0E003
-:10087000AA0DBB1D8C93A5E6B0E08C918F5F8C9385
-:1008800008F4E9CF80E080936500009165000A31AB
-:1008900010F009F019C0A5E6B0E0AD90BB24A3E8C4
-:1008A000B0E0AA0DBB1D8C91A42DB52D8C9381E0D9
-:1008B00030D37AD0229681E028D3A5E6B0E08C919F
-:1008C0008F5F8C9308F4E1CF8BE1809365000091FA
-:1008D00065000E3110F009F04CC080918000A42D0D
-:1008E000B52D8C9381E015D35FD0229681E00DD396
-:1008F00080918000A42DB52D8C9381E00AD354D033
-:10090000229681E002D3A5E6B0E0AD90BB24A3E837
-:10091000B0E0AA0DBB1D8C9180938000A5E6B0E0ED
-:10092000AD90BB24A3E8B0E0AA0DBB1D4C91AFE78E
-:10093000B0E0AD90BB24A3EAB0E0AA0DBB1D0C91C2
-:10094000041709F405C0ABE6B0E08C9181608C938C
-:10095000AFE7B0E08C918F5F8C93AFE7B0E08C9104
-:1009600087708C93A5E6B0E08C918F5F8C9308F430
-:10097000AECF80E080936500009165000F3310F0EA
-:1009800009F011C08DE0A42DB52D8C9381E0C1D26A
-:100990000BD0229681E0B9D2A5E6B0E08C918F5FB2
-:1009A0008C9308F4E9CF0895AFE7B0E0AD90BB2495
-:1009B000A3EAB0E0AA0DBB1D0C91A881B9814C91AE
-:1009C0000427AFE7B0E0AD90BB24A3EAB0E0AA0DE6
-:1009D000BB1D0C93AFE7B0E0AD90BB24A3EAB0E041
-:1009E000AA0DBB1D8C91A3EDB0E08D9388278D934C
-:1009F0008D938C9394E0A3EDB0E0ACD2A3EDB0E086
-:100A0000AD90BD90ABEAB0E0AA0DBB1D8C91809378
-:100A10008100AFE7B0E0AD90BB24A3EAB0E0AA0D3F
-:100A2000BB1D0C911127222733274FE050E060E0D7
-:100A300070E047D2A3EDB0E00D931D932D933D934D
-:100A4000A3EDB0E00D911D912D913C9140E150E05E
-:100A500060E070E0A4D2A3EDB0E00D931D932D9360
-:100A60003D93A3EDB0E0AD90BD90ABEAB0E0AA0D30
-:100A7000BB1D8C9180938200009182004091810087
-:100A8000040FA2E8B0E00C93A2E8B0E08C9180954E
-:100A90008C93A2E8B0E08C91A3EDB0E08D93882711
-:100AA0008D938D938C9391E0A2E8B0E038D297E0DB
-:100AB000A3EDB0E04FD2009182004091D300042B0F
-:100AC000A2E8B0E00C9300918200A881B9814C911A
-:100AD000040FA2E8B0E00C93A2E8B0E08C91A3ED83
-:100AE000B0E08D9388278D938D938C9391E0A2E84D
-:100AF000B0E015D297E0A3EDB0E02CD200918200D7
-:100B00004091D300042BA2E8B0E00C93A2E8B0E03F
-:100B10008C91A3EDB0E08D9388278D938D938C936A
-:100B200094E0A2E8B0E00ED294E0A3EDB0E0FFD1F3
-:100B3000009182004091D300042BA2E8B0E00C9316
-:100B4000AFE7B0E08C918F5F8C93AFE7B0E08C9112
-:100B500087708C93AFE7B0E0AD90BB24A3EAB0E020
-:100B6000AA0DBB1D0C91409182000427AFE7B0E0B5
-:100B7000AD90BB24A3EAB0E0AA0DBB1D0C93089571
-:100B800081EE80936800A6E6B0E08D919C91ABED7C
-:100B9000B0E08D939C93A3D1CDD281E08A9382E083
-:100BA0008A9388E08A9381E08A93A8E6B0E0E7E33D
-:100BB00086E0E8940ED200916B00003009F03BC053
-:100BC00080E08093650000916500073010F009F027
-:100BD00031C0A5E6B0E0AD90BB24A3EAB0E0AA0D19
-:100BE000BB1DBA93AA938EDB2296A6E6B0E08D9148
-:100BF0009C91ABEDB0E08D939C9371D19BD281E041
-:100C00008A9382E08A9388E08A9381E08A93A5E6BA
-:100C1000B0E0AD90BB24A3EAB0E0AA0DBB1DE7E3B2
-:100C200086E0E894D6D1A5E6B0E08C918F5F8C93F6
-:100C300008F4C9CF2BC080E0809365000091650067
-:100C4000073010F009F022C08FEF80936800A6E60D
-:100C5000B0E08D919C91ABEDB0E08D939C933FD132
-:100C600069D281E08A9382E08A9388E08A9381E066
-:100C70008A93A8E6B0E0E7E386E0E894AAD1A5E687
-:100C8000B0E08C918F5F8C9308F4D8CF3FD008955B
-:100C900080E080936500A881B9814C910091650046
-:100CA000041710F009F02BC0ADD0A8E6B0E05DD17C
-:100CB0000C92A5D0A8E6B0E0BA93AA9323DB2296C3
-:100CC000A6E6B0E08D919C91ABEDB0E08D939C9346
-:100CD00006D130D281E08A9382E08A9388E08A93B9
-:100CE00081E08A93A8E6B0E0E7E386E0E89471D17A
-:100CF000A5E6B0E08C918F5F8C9308F4CCCF009187
-:100D00006E00013009F001C001D0089586EF809394
-:100D10006800A6E6B0E08D919C91ABEDB0E08D93BC
-:100D20009C93DDD007D281E08A9382E08A9388E0A9
-:100D30008A9381E08A93A8E6B0E0E7E386E0E8944E
-:100D400048D18FEF80936800A6E6B0E08D919C912A
-:100D5000ABEDB0E08D939C93C2D0ECD181E08A934F
-:100D600082E08A9388E08A9381E08A93A8E6B0E0E3
-:100D7000E7E386E0E8942DD108953FFA11250500B8
-:100D800001B0023B364D590280817C00000000001A
-:100D9000000000000000000000000000788E17CE68
-:100DA0007B5AD52D9B0078800B7750DA8598DF0031
-:100DB0007AD8283A4E454F373533303A292020210A
-:100DC000484156452046554E21207A80202020203B
-:100DD0002020202020202020202020202020202013
-:100DE000202020207AD854484953204348414E4E71
-:100DF000454C20495320424C4F434B4544208E2EB6
-:100E00009F2E0895E82DF92D089500FC05D0089532
-:100E100030D005D001F8F9CF37D03FD00895842FD6
-:100E2000952FA62FB72FE894452B462B472B11F46F
-:100E300068940895402F512F622F732F0027112798
-:100E400022273327F0E2440F551F661F771F001F2C
-:100E5000111F221F331F081B190B2A0B3B0B41606C
-:100E600028F4080F191F2A1F3B1F4E7FFA9559F7C8
-:100E700008950024E89477FF02C006D0689400F833
-:100E800009D010F80124089512D04F5F5F4F6F4FC3
-:100E90007F4F0895E89437FF0895009510952095A9
-:100EA00030950F5F1F4F2F4F3F4F68940895409527
-:100EB00050956095709508950427152726273727A4
-:100EC0000895042315232623372308953197F1F736
-:100ED0000895689462F80895E89462F808958D91F1
-:100EE00087FB16F48FEF01C088278D938C930895AC
-:100EF0000D911D912D913C9108953C932E931E933D
-:100F00000E930895481A590A0895481A8827580ACE
-:100F1000089599275A924A92480E591E0895903082
-:100F200029F08C91880F9A95E9F78C930895903069
-:100F300041F0DEDF000F111F221F331F9A95D1F7FA
-:100F4000DCCF0895903029F08C9186959A95E9F739
-:100F50008C930895903041F0CBDF36952795179507
-:100F600007959A95D1F7C9CF0895C8953196002075
-:100F7000089591E0882321F08894991F8A95E9F7D4
-:100F8000892F909508950F938F9308E09927881FD4
-:100F900008F493950A95D9F78F910F910895040F4E
-:100FA000151F261F371F0895041B150B260B370B23
-:100FB0000895E199FECFEEBBFFBB3196089591E015
-:100FC00001C092E0F6DFE09A8DB38D939A95D1F748
-:100FD0000895CFDFFF27008009220082E39500807B
-:100FE00009220082082F692FEA9549918D91188175
-:100FF0001730B9F41981103031F4806880952A8156
-:10100000285F992724C0113041F4BDDF90FD80682E
-:1010100080952A81275F99271AC0B5DF90FF8068E5
-:10102000F7CF1981103029F480952A81275F9927FD
-:101030000EC0113039F4A7DF80952A81265F909584
-:10104000917005C0A0DF80952A81265F9170089479
-:1010500016F028F401C018F00080002A03C00080B8
-:101060000622000000825BD05AD0969587952A957B
-:1010700079F74A9509F0BACF239651D0089579DFD0
-:10108000FF27008009220082E39500800922008268
-:101090003297082F692F4991342F1881173049F45E
-:1010A0001981103019F42A81295F0BC02A81285F29
-:1010B00008C01981103019F42A81285F02C02A81E2
-:1010C000275F522F0080002216F019F0FBCF09F4A1
-:1010D000F9CF25D024D023D088940080002209F0B5
-:1010E00008942A9519F087959795F4CF5A3028F0EF
-:1010F0005950991F881F5A95E1F70EF48095188171
-:10110000173009F48F778D93303031F48D3029F614
-:1011100088278C93239608954A95E1F3BECF8F9349
-:101120009F938091DB009091DC000197F1F79F91F4
-:101130008F91089500E519E726E330E0ABEDB0E0CC
-:101140004D915D916D917C9163DEABEDB0E04D937F
-:101150005D936D937D93ABEDB0E00D911D912D915D
-:101160003C914AE250E060E070E01EDFABEDB0E0A1
-:101170000D931D932D933D93ABEDB0E00D911D911B
-:101180002D913C9148E050E060E070E041DEABED35
-:0C119000B0E04D935D936D937D93089546
-:00000001FF
diff '-x=.git' -uNr hacktv/cards/funcard/syster-extra-cmds.txt hacktv_patched/cards/funcard/syster-extra-cmds.txt
--- hacktv/cards/funcard/syster-extra-cmds.txt	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/cards/funcard/syster-extra-cmds.txt	2020-07-13 13:21:25.970222200 +0200
@@ -0,0 +1,18 @@
+Switching between Crypt-Modes:
+
+0x0400: DES-Mode
+0x0401: Passthrough-mode (first 8 Bytes from ECM will returned back as CW)
+0x0402: XTEA-Mode (ECM and Signature has to be passed)
+
+Switching ATR:
+0x1400: Premiere DE
+0x1401: C+ France
+0x1402: C+ Poland
+0x1410: Premiere DE + Date-Checking
+0x1411: C+ France + Date-Checking
+0x1412: C+ Poland + Date-Checking
+
+
+Update DES-Key:
+
+24 0Y XX XX XX XX XX XX XX XX = Update DES-Key 0Y 00...0F
diff '-x=.git' -uNr hacktv/cards/funcard/syster.eep hacktv_patched/cards/funcard/syster.eep
--- hacktv/cards/funcard/syster.eep	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/cards/funcard/syster.eep	2020-07-13 13:21:25.974212400 +0200
@@ -0,0 +1,10 @@
+:10000000C4A5A8187493C76500000000000012344E
+:1000100000E2516D1597515500AE529049F1F1BB78
+:1000200000E9EBB3A6DB3C87000000000000000005
+:1000300000000000000000000000000000000000C0
+:1000400000000000000000003322110077665544D4
+:10005000BBAA9988FFEEDDCC714078D5109190480D
+:100060007A0C26019D9E57D50001FFFF606ADFC113
+:1000700021BC0001FFFF616BDFBB21800102190180
+:090080001A011B011C0100100013
+:00000001FF
diff '-x=.git' -uNr hacktv/cards/funcard/syster.hex hacktv_patched/cards/funcard/syster.hex
--- hacktv/cards/funcard/syster.hex	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/cards/funcard/syster.hex	2020-07-13 13:21:25.979199800 +0200
@@ -0,0 +1,337 @@
+:1000000043C152C151C150C1BCC64EC14DC181C6D0
+:100010004BC14AC149C148C147C1D5C6F6C6F5C69C
+:10002000E6C6F3C6F2C6E3C6F0C6EFC6E0C6EDC646
+:10003000ECC6DDC6EAC6E9C6DAC6E7C6E6C6D7C676
+:10004000E4C6E3C6D4C6E1C6E0C6D1C6DEC6DDC698
+:10005000CEC6DBC6DAC6CBC6D8C6D7C6B1C6ECC6D6
+:100060001EC71DC71CC708C71AC719C705C717C7AA
+:1000700016C702C714C713C7FFC611C710C7FCC6EF
+:100080000EC70DC7F9C60BC70AC7F6C608C707C70C
+:10009000F3C605C704C7F0C602C701C7DDC6010223
+:1000A0004F53304274724B1D000102004000007437
+:1000B000724B1D0001024F53304074724B1D00A063
+:1000C000021CE00C01FF14E1E500A0021838120048
+:1000D000FF14808300A0021C381405FF14E1E50022
+:1000E00007030602050104000400050106020703D8
+:1000F00000030201040506073112503313214200A8
+:100100005152304353702203736241602320020135
+:1001100061634032101171721F0001020344030435
+:10012000050607680708090A0B8C0B0C0D0E0FB0AB
+:100130000F10111213D41314151617F81718191AD3
+:100140001B1C1B1C1D1E1F201C1F180A120E071A29
+:1001500004150B100C1B0F09141E050D171D081399
+:100160003E332C25393038263C342D29362B3A3174
+:10017000243D3B3F28352F321FB028EBD10D427E66
+:10018000C5599334A66AFC87B0E3177D2B96DE48E9
+:100190000A346C81C55FA9F22ED072B7950C48EB7A
+:1001A000536AC914AFF1368D8D4EB1E86B3517D255
+:1001B000F093562F0CCAA974B24FD4180BF67E25B3
+:1001C000C13C6A83AD5097E9E9B442273ECB85181C
+:1001D000560A9F70F1AD6CD335E05B0D68D3967A0B
+:1001E000F92EC2B11F84AC476B1C0DA3D67A30C563
+:1001F00084F1BE58E92F4792D134BDE38B58429E1B
+:100200007AAFC0052CF61769B4D7E3485E218D722A
+:1002100009603FA695CBFA1C822714CAF9906F5C3F
+:10022000EBD87DA34E35B1065C906FF9354E822731
+:1002300006EBCA14A3D87DB152F86F169CCB09A562
+:10024000ED273A8143B4D07E2E95B26F7906C7F878
+:100250004BE0D13CA45A1D830CE27B18904DC7B1F2
+:10026000638FDE2539F6A45AF217854E5CB02BED6C
+:10027000A47938936FCAD106010202020202020178
+:10028000020202020202010011241FBECFE5D2E0E9
+:10029000DEBFCDBF20E0A0E6B0E001C01D92AF39C7
+:1002A000B207E1F7CDD829C9ABCE0F931F93CF93F7
+:1002B000DF9304E710E0C0E0D0E0E02FF12F608191
+:1002C0000E5F1F4F8C2F9D2F82589F4F05D921966F
+:1002D000C830D10591F7DF91CF911F910F9108950B
+:1002E000EF92FF920F931F93CF93DF9394E7E92E42
+:1002F00090E0F92EC82FD0E023E0CC0FDD1F2A9527
+:10030000E1F700E010E0EE2DFF2D6081F2E0EF0E4E
+:10031000F11C802F912F8C0F9D1F885F9F4FDCD881
+:100320000F5F1F4F0830110571F7DF91CF911F91BB
+:100330000F91FF90EF9008951F93CF93DF93CDB768
+:10034000DEB769970FB6F894DEBF0FBECDBF162F8C
+:1003500012BE94E093BF613129F448E050E060E0C0
+:1003600070E011C060916D006F7070E0660F771FD4
+:10037000680F711D23E0660F771F2A95E1F7685F0C
+:100380007F4F48E050E085E690E086D8E4E7F0E073
+:10039000AC2FBD2F11968A2F9B2F208132962D9343
+:1003A00020E0E439F207C9F74C2F5D2F4F5E5F4F15
+:1003B00065E670E080D7482F592FEC2FFD2F7196FE
+:1003C000A6E7B0E08C2F9D2F4996219130E02D9328
+:1003D0003D93E817F907C9F790916D00907F903130
+:1003E000C9F41131B9F08091600090916100481713
+:1003F000590768F080916200909163008417950717
+:1004000030F0898D181303C01092640003C081E09E
+:100410008093640013BE69960FB6F894DEBF0FBEDA
+:10042000CDBFDF91CF911F9108952F923F924F92B0
+:100430005F926F927F928F929F92AF92BF92CF9274
+:10044000DF92EF92FF920F931F93CF93DF93CDB77D
+:10045000DEB769970FB6F894DEBF0FBECDBF0C2F85
+:100460001D2F0F5F1F4FE02EF12E00E010E0282F10
+:10047000217030E0C22ED32ECC0CDD1CCC0CDD1C48
+:100480008C2D9D2D800F911F880F991F880F991F0C
+:10049000885B9F4F1CD8EE2DFF2D61937193819344
+:1004A0009193EE2EFF2E0F5F1F4F0430110541F781
+:1004B000ECE7F0E024E730E0198A1A8A1B8A1C8AEC
+:1004C00040E050E060E070E0C12CD12CE12CF12C38
+:1004D00080E090E0A0E0B0E08824942EA52EB62E17
+:1004E000729162917727462F572F60E070E048297C
+:1004F00059296A297B2949885A886B887C88B62CB7
+:10050000A52C942C88241089482C592C6A2C7B2CDF
+:10051000412A498A5A8A6B8A7C8ABA2FA92F982F36
+:1005200088271085812BFE2CED2CDC2CCC24108D03
+:10053000C12A2E173F0781F620916F00223009F063
+:10054000D1C0509064005D8A812C912CA12CB12CDB
+:10055000212C312C442E552E662E772E24E0440C6F
+:10056000551C661C771C2A95D1F7042F152F262FB2
+:10057000372FE5E03695279517950795EA95D1F73A
+:100580004026512662267326440E551E661E771E8F
+:100590003B2D2A2D192D082D0370112722273327D3
+:1005A000000F111F000F111FE1E0F0E0EC0FFD1F25
+:1005B000E00FF11F0081118122813381080D191D87
+:1005C0002A1D3B1D0425152526253725800F911F43
+:1005D000A21FB31F07E4801A06E8900A08ECA00ADD
+:1005E00001E6B00A3B2D2A2D192D082DEBE036959A
+:1005F000279517950795EA95D1F7037011272227BC
+:100600003327202F312F220F331F220F331FE1E01A
+:10061000F0E0EC0FFD1FE20FF31F0081118122813A
+:100620003381482C592C6A2C7B2C400E511E621EA3
+:10063000731E4E8A5F8A688E798E482E592E6A2ED6
+:100640007B2E14E0440C551C661C771C1A95D1F7C0
+:10065000082F192F2A2F3B2FE5E036952795179560
+:100660000795EA95D1F70425152526253725080F86
+:10067000191F2A1F3B1F4E885F88688C798C0425C0
+:10068000152526253725400F511F621F731F07E0D0
+:1006900020163104E1F4C816D906EA06FB0651F427
+:1006A00049885A886B887C88441655066606770602
+:1006B00069F004C0C82ED92EEA2EFB2E81E080936B
+:1006C0006400BF2DAE2D9D2D8C2D0CC01D8A0FEF0B
+:1006D000201A300A10E22116310409F03BCF2D898F
+:1006E00020936400E6E7F0E020E030E0C42ED52E51
+:1006F000E62EF72E022E04C0F694E794D794C79402
+:100700000A94D2F7DD24C192D192C82ED92EEA2EB6
+:10071000FB2E022E04C0F694E794D794C7940A9453
+:10072000D2F7DD24D782C682285F3F4F20323105C1
+:10073000E9F669960FB6F894DEBF0FBECDBFDF9124
+:10074000CF911F910F91FF90EF90DF90CF90BF90CE
+:10075000AF909F908F907F906F905F904F903F9061
+:100760002F900895CF92DF92EF92FF920F931F93F5
+:10077000CF93DF93082F192FC0917200D09173008F
+:10078000EE24E394F12CC0907000D0907100802F83
+:10079000912F8E0D9F1D93D6EC2FFD2FEC0DFD1D7F
+:1007A000EF70FF27EE0FFF1FEC58FF4F90E0918393
+:1007B00080838FEFC81AD80AD0927100C09270005F
+:1007C0008FEFE81AF80A8BE0E816F104E1F6EC2F57
+:1007D000FD2FEE0FFF1FEC58FF4F8081918191603C
+:1007E000918380832996CF70DD27CC0FDD1FCC58F5
+:1007F000DF4F88819981916099838883802F912F21
+:100800005ED690E09160DF91CF911F910F91FF90A4
+:10081000EF90DF90CF9061C2CF93DF932091720071
+:1008200030917300609170007091710041E050E070
+:10083000C62FD72FC20FD31F2197E82FF92FE40F10
+:10084000F51FC895E02DAC2FBD2FA40FB51FAF70BD
+:10085000BB27AA0FBB1FAC58BF4FF0E0ED93FC9332
+:100860004F5F5F4F4B30510541F7665F7F4F70938D
+:10087000710060937000E22FF32FEE0FFF1FEC5812
+:10088000FF4F40815181516051834083E22FF32F0C
+:100890003996EF70FF27EE0FFF1FEC58FF4F2081B6
+:1008A0003181316031832083E82FF92FC895802D65
+:1008B00090E09160DF91CF9110C2EF92FF920F9381
+:1008C0001F93CF93DF93C82FD92FC33084E1D8076C
+:1008D00008F045C0C115E4E1DE0708F0ACC0C23045
+:1008E000F5E0DF0748F5C11585E0D80708F044C1F9
+:1008F000C115E2E0DE0709F47DC050F4DA952297D5
+:1009000008F0D0C181E091E0E8D1C4E7D0E05EC05A
+:10091000C13082E0D80709F485C08C2F9D2F9450F8
+:10092000039708F0BFC18FEF91E0D7D1C0936F005C
+:100930006C2F88E890E087C0C131E6E0DE0709F45B
+:1009400028C18C2F9D2FC231F6E0DF0710F496509E
+:1009500002C080529640039708F41BC1A3C1C23065
+:1009600087E5D80709F497C0E8F4C031F4E2DF075F
+:1009700068F4C11584E2D80708F06CC08C2F9D2F55
+:1009800080519441039708F456C08CC1C115E7E526
+:10099000DE0709F47DC0C130D74509F083C189EA7B
+:1009A00090E039C0C1158FE5D80709F477C028F465
+:1009B000DE55239708F4D3C075C1C330FFE5DF07C8
+:1009C00008F4CDC0219609F446C16CC17BD199834E
+:1009D000888378D19B838A8381E091E07ED12496BD
+:1009E000E0E0C438DE0791F760DC6CD16BD180E0C9
+:1009F00091E05AC180916D008F70813051F030F0DC
+:100A0000823009F058C18FEB90E005C085ED90E091
+:100A100002C08AEC90E0DF91CF911F910F91FF907F
+:100A2000EF90FACE8CE790E0DF91CF911F910F917C
+:100A3000FF90EF9097CE8FEF91E04FD1C0936D0074
+:100A40006C2F87E890E0DF91CF911F910F91FF907D
+:100A5000EF9042C58FEF91E040D1C0936E00C4E7A4
+:100A6000D0E030D1998388832DD19B838A8384E21F
+:100A700091E033D12496F0E0C438DF0791F78091FC
+:100A80006E00DF91CF911F910F91FF90EF9028CCD6
+:100A900084EB90E0C0CF8EE990E0BDCF81E091E0A3
+:100AA0001CD110D1C82FD92F0DD1C130D105B9F12A
+:100AB00040F0C230D10509F457C0239709F454C05F
+:100AC000FAC081E091E009D1109273001092720097
+:100AD000109271001092700006E710E0C1E0D0E0C3
+:100AE0008C2F9D2F8E589F4FEAD490E0E02FF12F4E
+:100AF000819391930E2F1F2F2196CA30D10581F734
+:100B000082E091E0909375008093740080E091E022
+:100B100090938900809388008BE090E097C081E0FB
+:100B200091E0DBD01092730010927200109271006D
+:100B30001092700006E710E0C1E0D0E08C2F9D2FEE
+:100B400088599F4FBCD490E0E02FF12F819391936F
+:100B50000E2F1F2F2196CA30D10581F7D1CF81E00A
+:100B600091E0BBD0AFD0AED081E091E0B6D0AAD0BA
+:100B7000A9D08AE491E098C081E091E0AED0C0E2D3
+:100B8000D0E0A0D09FD081E091E0A7D02197C9F715
+:100B900092C081E091E0A1D08C2F9D2F807F99277A
+:100BA00025E0969587952A95E1F780936E0010923F
+:100BB000730010927200109271001092700004E79E
+:100BC00010E034E9E32E30E0F32E7CD0E02FF12F5B
+:100BD0009183808377D0E02FF12F9383828381E00C
+:100BE00091E07BD00C5F1F4FE016F10671F780910A
+:100BF0006F00811105C06C2F80916E009DDB03C0DA
+:100C000080916E0012DC80916400811113C086E037
+:100C100091E0909375008093740082E091E090934E
+:100C200087008093860010927300109272008AE011
+:100C300090E00CC08AE091E090937500809374007E
+:100C4000109273001092720081E090E09093710016
+:100C50008093700030C04091700050917100141664
+:100C6000150604F52091720030917300E22FF32FE6
+:100C7000EE0FFF1FEC58FF4F808191812F5F3F4F98
+:100C800030937300209372004150510950937100CA
+:100C9000409370002031310539F4109273001092A6
+:100CA000720002C081E091E0DF91CF911F910F911E
+:100CB000FF90EF9012C0DF91CF911F910F91FF90A5
+:100CC000EF90089580919C008823E1F380919D002E
+:100CD00090919E0010929C000895209199002130DF
+:100CE000E1F390939B0080939A0081E080939900B8
+:100CF00008951FBC89E08EBD85E890E09BBD8ABD4C
+:100D000082E489BFC698BE9810929C001092990008
+:100D100008951F920F920FB60F9211242F933F93B5
+:100D20004F935F936F937F938F939F93AF93BF93F3
+:100D3000EF93FF9313BE7894809194008F5F80931C
+:100D4000940080919400813039F410929400BADFBD
+:100D5000B9DF81E091E0C1DFF89412BE84E083BF87
+:100D6000FF91EF91BF91AF919F918F917F916F9183
+:100D70005F914F913F912F910F900FBE0F901F9059
+:100D800018951F920F920FB60F9211242F933F9335
+:100D90004F935F938F939F93AF93BF93EF93FF9383
+:100DA00020919800822F90E027FF3AC0E82FF92F7A
+:100DB000E058F109E232F10540F5E35FFF4F099495
+:100DC00080E88093980080919900811103C010926F
+:100DD000980068C080919A0090919B00880F991F9D
+:100DE0009460909397008093960010929900809160
+:100DF000960090919700282F21702093950096954A
+:100E00008795909397008093960080919500882312
+:100E100011F0BE9801C0BE9A809198008F5F19C0F2
+:100E200036B330743093950080329105C0F5E82FC9
+:100E3000F92FE15DFF4F099480919900813011F401
+:100E400080E807C0B6992EC01092970010929600C5
+:100E500081E08093980026C080919C008111B7CFDB
+:100E6000809196009091970090939E0080939D00B2
+:100E700081E080939C00ABCF809196009091970089
+:100E800096958795311103C040E050E002C040E0E4
+:100E900051E0842B952B90939700809396002F5FC1
+:100EA00020939800FF91EF91BF91AF919F918F9107
+:100EB0005F914F913F912F910F900FBE0F901F9018
+:100EC00018959F92AF92BF92CF92DF92EF92FF92CE
+:100ED0000F931F93CF93DF93CDB7DEB728970FB64D
+:100EE000F894DEBF0FBECDBFE82FF92F98E0AC2FEE
+:100EF000BD2F119601900D929A95E1F7062F172FAD
+:100F0000AC2FBD2F1196EC2EFD2E87E0E81AF108CC
+:100F1000C42ED52EE8E0CE0ED11C842F952FE42FC1
+:100F2000F52FC895202D2370E82FF92F09F1C895CA
+:100F3000902CE02FF12F2781322F330FE1E0F0E0EA
+:100F4000EC0FFD1FE90DF11D20812170232BE02FF7
+:100F5000F12F2783E82FF92FC895202DE1E0F0E04D
+:100F6000EC0FFD1FE20FF11D2081269520831EC08E
+:100F7000C895202DA62EB72EA20EB11CC895202DE7
+:100F8000E62FF72FE20FF11D2081269517963C9151
+:100F9000179730FD02C0E0E001C0E0E82E2BEA2DFB
+:100FA000FB2D2083369517963C93179701968C1549
+:100FB0009D0509F0B4CF015011091197EA16FB06FF
+:100FC00009F0ABCF28960FB6F894DEBF0FBECDBFA9
+:100FD000DF91CF911F910F91FF90EF90DF90CF9015
+:100FE000BF90AF909F900895CF92DF92EF92FF92C3
+:100FF0000F931F93CF93DF93C82FD92F88E0E42F4F
+:10100000F52F11928A95E9F7A42FB52F2AE230E047
+:1010100040E050E0022F132F0B5F1F4F8C91880F81
+:101020008C93E02FF12FE41BF50BEC0FFD1FC895FF
+:10103000902DEC2FFD2FC895F02CFFE1FF1611F03D
+:10104000EFEF01C0EFE1DE2ED9229D2D969596950A
+:101050009695E62EF72EE90EF11CEE2DFF2DE08081
+:10106000F12CED2DE770CE2CDF2C02C0D594C79467
+:10107000EA95E2F7C0FE02C081608C934F5F5F4F3C
+:101080004630510559F61196265031092A3F8FEF07
+:10109000380709F0BDCFDF91CF911F910F91FF90DD
+:1010A000EF90DF90CF900895CF93DF93C62FD72F87
+:1010B00040E0E82FF0E0E858FD4F262F372F2D5F56
+:1010C0003F4FC895802D4817B8F5AC2FBD2F8C9198
+:1010D000982F969511968C911197879588278795CB
+:1010E000892B8D9313968C911397982F9695149620
+:1010F0008C911497879588278795892B13968C93C5
+:1011000013972A173B0719F7888186958695869548
+:101110008795882787959B819695892B8B838C81D2
+:101120008695869586958795882787959F8196953C
+:10113000892B8F834F5FC5CFDF91CF9108956F9239
+:101140007F928F929F92AF92BF92CF92DF92EF9257
+:10115000FF920F931F93CF93DF93CDB7DEB76497C2
+:101160000FB6F894DEBF0FBECDBFE82EF92E062FC6
+:10117000172FB12C88248394912CC62ED72E84E06F
+:10118000C80ED11C4C2F5D2F4F5F5F4F6E2D7F2DF2
+:1011900088E491E029DF4C2F5D2F475F5F4F602F80
+:1011A000712F88E191E020DF6C2F7D2F6F5F7F4FE3
+:1011B000AC2FBD2F199640E051E02FE1722EE62FA3
+:1011C000F72F91916E2F7F2F8D918927982F9F73E5
+:1011D0009695E42FF52FFF27E92BE858FE4FC89589
+:1011E000A02C80FF03C0A29490EFA9228CEF682E60
+:1011F000670CE72DF0E0E850FF4FC895802D837015
+:10120000C895E02DE295E770282D392D01C0220FF9
+:10121000EA95EAF790E0A7FE0AC02095E1E1F0E048
+:10122000EC0FFD1FE80FF91F3081232308C0E1E117
+:10123000F0E0EC0FFD1FE80FF91F3081232BE1E1F7
+:10124000F0E0EC0FFD1FE80FF91F2083AA0C7A9441
+:101250007610CFCF40525109FFEF7F12B0CFAC2FA5
+:10126000BD2F5196E02FF12F9C91848189278D937A
+:10127000908194838193CE16DF06B1F76E2D7F2D7A
+:101280008B2D12DFB39480E1B8127CCF64960FB639
+:10129000F894DEBF0FBECDBFDF91CF911F910F91AC
+:1012A000FF90EF90DF90CF90BF90AF909F908F9086
+:1012B0007F906F9008955F926F927F928F929F922E
+:1012C000AF92BF92CF92DF92EF92FF920F931F9354
+:1012D000CF93DF93CDB7DEB7A0970FB6F894DEBFFC
+:1012E0000FBECDBF862E972E042F152FA82EB92EF8
+:1012F000CC2EDD2E89E1C80ED11CAC2FBD2F11964E
+:10130000EA2EFB2E712C6A2E5F2C40EF50E06C2FE2
+:101310007D2F675F7F4F882D992DD3DD8D85829539
+:10132000807F898748EE50E06C2F7D2F6F5E7F4F66
+:101330008A2D9B2DC6DD6C2F7D2F6F5E7F4F8C2FEE
+:101340009D2F0996FCDE40EE50E0662D752D8C2F0A
+:101350009D2F4196B6DD711004C08F81E02FF12FD3
+:10136000808729813A812F3FFFEF3F0711F42B81BE
+:101370003C8184E0EE2DFF2DAC2DBD2D01900D9212
+:101380008A95E1F7739488E0A80EB11C94E0C90E29
+:10139000D11CA4E0EA0EF11CB2E07B12B6CFFE8DA8
+:1013A000EF8D48A1898D9A8D6B8D5C8D550F5E7187
+:1013B000762F771F7727771F572B7D8DA02FB12F83
+:1013C0007C931196FC9311971296EC931297742FBD
+:1013D0007F7713967C931397782F770F441F4427BA
+:1013E000441F472B14964C931497492F440F881F82
+:1013F0008827881F842B15968C931597660F991F45
+:101400009927991F962B16969C93169717965C937F
+:10141000822F932FA0960FB6F894DEBF0FBECDBFDC
+:10142000DF91CF911F910F91FF90EF90DF90CF90C0
+:10143000BF90AF909F908F907F906F905F90089536
+:1014400058DC789488E890E03AD080936F0087E881
+:1014500090E035D080936D0042E050E06AE770E0A4
+:1014600080E690E019D042E050E06EE670E082E65F
+:1014700090E012D0C0E0D0E025DC082F192FD0FD7D
+:1014800003C0C02FD12FF8CF90FDFBCF9C2F882712
+:10149000802B912B12DAF5CFA82FB92F862F972FFB
+:1014A000E82FF92FE199FECF06C0FFBBEEBBE09A13
+:1014B00031960DB20D9241505040B8F70895E19920
+:1014C000FECF9FBB8EBBE09A99278DB30895A6E10E
+:1014D000B0E044E050E0E4CF262FE199FECF9FBB7F
+:1014E0008EBBE09A01970DB2021631F02DBB0FB6FC
+:0E14F000F894E29AE19A0FBE0895F894FFCFA7
+:00000001FF
diff '-x=.git' -uNr hacktv/cards/funcard/vc.eep hacktv_patched/cards/funcard/vc.eep
--- hacktv/cards/funcard/vc.eep	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/cards/funcard/vc.eep	2020-07-13 13:21:25.984189300 +0200
@@ -0,0 +1,25 @@
+:1000000065E7711AB488D77628D04C6E868CC843C1
+:10001000A9EC604205F23D1C6CBCAFC32BB5DC9073
+:10002000F905EA51469DE26070526726614942092E
+:10003000509990A2360EFD39D94508DB7CF956F768
+:100040005818225438CD3D9409E68E0D9A86FC1C32
+:10005000A0198FBCFD8DD15756F2B64FC9BD2AB33A
+:100060009D815DE005B5B926673C65A0BA39C7AF8B
+:10007000332447A6201E146F489B4DA6F9D9DF6E86
+:10008000AC84FA8B2EB67619C1B0A3BB0CFD70728E
+:10009000CA55EFA07FBF59AD91619D53B327D5D904
+:1000A0000F59A66F73FB994CFB45548E205FB3B17B
+:1000B00038D06BA74039ED2ADA438D5192D6E361EF
+:1000C000658C71E68465870355BC6407BB799E40E7
+:1000D0009789C4148F8B414D2AAAE8E108CD824349
+:1000E0008F6F369B7247F2A449DD8B6E26C6BFB771
+:1000F000D844C370A34CB6B2379B09DF322824869C
+:100100008D5FE64B5DD02FDBAC2E781ECC52C161EB
+:10011000EA82CAB3F48F638E6CBCAFC32BB5DC909C
+:10012000F905EA51469DE260013559790000550F05
+:1001300000000000106E1CBDFE44EB79F3AB5D23A4
+:10014000B320D2E7FC00036FD8B7F7F3557247131B
+:100150007B0C08018A2C70560A85181443C9466422
+:100160006C9A99590A6C40D517B32C6941E8E70E8F
+:100170003322110077665544BBAA9988FFEEDDCC87
+:00000001FF
diff '-x=.git' -uNr hacktv/cards/funcard/vc.hex hacktv_patched/cards/funcard/vc.hex
--- hacktv/cards/funcard/vc.hex	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/cards/funcard/vc.hex	2020-07-13 13:21:25.988174600 +0200
@@ -0,0 +1,249 @@
+:10000000C8C0E3C0E2C0E1C01BC6DFC0DEC0DDC0C7
+:10001000DCC0DBC0DAC0D9C0D8C034C655C654C6AF
+:1000200045C652C651C642C64FC64EC63FC64CC64E
+:100030004BC63CC649C648C639C646C645C636C67E
+:1000400043C642C633C640C63FC630C63DC63CC6A0
+:100050002DC63AC639C62AC637C636C610C64BC6DE
+:100060007DC67CC67BC667C679C678C664C676C6BA
+:1000700075C661C673C672C65EC670C66FC65BC6FD
+:100080006DC66CC658C66AC669C655C667C666C61A
+:1000900052C664C663C64FC661C660C63CC6900001
+:1000A0007AD854484953204348414E4E454C204944
+:1000B0005320424C4F434B4544207A80202020203F
+:1000C0002020202020202020202020202020202030
+:1000D000202020207AD8214B455955504441544581
+:1000E00021202020202020202020202020207AD8FD
+:1000F00020534B59392D4D4F44452020204841561F
+:10010000452046554E203A297AD820585445412D4D
+:100110004D4F4445202020484156452046554E200D
+:100120003A297AD820534B59372D4D4F444520203A
+:1001300020484156452046554E203A297AD820205D
+:100140005441432D4D4F44452020204841564520E1
+:1001500046554E203A297800000000000000007843
+:10016000800B7750DA8598DF788E17CE7B5AD52DA5
+:100170009B7C000000000000000000000000000068
+:1001800000003FFA1125050001B0023B364D59022F
+:10019000808111241FBECFE5D2E0DEBFCDBF10E0CD
+:1001A000A0E6B0E0E6E6FFE003C0C89531960D9208
+:1001B000AE36B107D1F721E0AEE6B0E001C01D9246
+:1001C000A238B207E1F7DCD5CCC61ACF982F9295AA
+:1001D0008927982F969596958927982F9695892700
+:1001E0008170089590916C00992391F0282F2A7ABC
+:1001F00026958575880F282B822F8C7C982F9695B5
+:100200009695822F8373880F880F892B829580950E
+:1002100008950F931F93CF93DF93A62FB72FE62F49
+:10022000F72F062F172F0A5F1F4F282F6081C62F29
+:10023000CF73D0E031814C2F5D2F40575F4F842F1B
+:10024000952F88569F4FC82FD92F9881C42FD52F0F
+:1002500088818927922F931B890F631B622770E087
+:1002600090E02FD63281832782838381980F9383F6
+:10027000220F211D275B32960E171F07B9F616961F
+:100280008C91169717966C91179770E090E019D69D
+:100290002C91820F821708F48F5F875C8C931196E4
+:1002A0008C911197980F981708F49F5F91571196AA
+:1002B0009C93DF91CF911F910F9108958F929F9200
+:1002C000AF92BF92DF92EF92FF920F931F93CF9363
+:1002D000DF93CDB7DEB728970FB6F894DEBF0FBE19
+:1002E000CDBF0C2F1D2F0F5F1F4FAC2EBD2E89E0F1
+:1002F000A80EB11CE02FF12F1192AE16BF06E1F748
+:100300002FE6822E20E0922E3AE8E32E30E0F32E04
+:10031000602F712FA82DB92D8D918A2E9B2E79DFFC
+:10032000E814F904A9F79EE8892E90E0992ED12CC3
+:10033000602F712F8D2D6DDF602F712F8D2D69DF57
+:10034000EE2DFF2DD190EE2EFF2E8885D81631F0A0
+:1003500080918F00816080938F0002C010928F0087
+:100360008E149F0429F780E4E82EF12C602F712F62
+:1003700080918E004EDFF1E0EF1AF108B9F7E1E667
+:10038000F0E0A02FB12F8D910A2F1B2F8193AA1679
+:10039000BB06B9F728960FB6F894DEBF0FBECDBFE7
+:1003A000DF91CF911F910F91FF90EF90DF90BF9061
+:1003B000AF909F908F9008952F923F924F925F92AF
+:1003C0006F927F928F929F92AF92BF92CF92DF9265
+:1003D000EF92FF920F931F93CF93DF9300D000D043
+:1003E000CDB7DEB79C838B83262E372E142F3A830E
+:1003F0002983A62FB72FED91FC91AE2FBF2F43E09D
+:10040000AA0FBB1F4A95E1F7A80FB91F8C91882E40
+:10041000812611969C9011971296AC901297139684
+:10042000BC9013971496CC9014971596DC90159762
+:100430001696EC90169717969C9117978C92A82D6C
+:10044000B92DAF70BB27A20FB31FA056BF4F282DE9
+:10045000392D4A2D5B2D6C2D7D2D8E2D04E044D53C
+:10046000422E532E29813A81240D351D20573F4FAE
+:100470009C91A22FB32F8C91890F8095880F811D9D
+:10048000180F1295110F111D3196E770FF27A22D3D
+:10049000B32DED93FC9353E0EE0FFF1F5A95E1F758
+:1004A000AB81BC81AE0FBF1F8C9118271C930F909E
+:1004B0000F900F900F90DF91CF911F910F91FF90B0
+:1004C000EF90DF90CF90BF90AF909F908F907F90F4
+:1004D0006F905F904F903F902F9008952F923F9292
+:1004E0004F925F926F927F928F929F92AF92BF9244
+:1004F000CF92DF92EF92FF920F931F93CF93DF93F0
+:10050000CDB7DEB7C754D1090FB6F894DEBF0FBE22
+:10051000CDBF482E592E6C2E7D2E21E4620E711C0B
+:10052000A62DB72D1D921C9280917000411451048C
+:1005300071F4843308F4EBC08B33B0F0843408F4E6
+:10054000E9C08B3468F078E4872E912C10C0833397
+:1005500008F4DDC08B3340F068E1862E912C07C093
+:1005600050E4852E912C03C048E0842E912CEC2F72
+:10057000FD2F31962E2E3F2E108211821282138271
+:10058000148215821682178238966E167F06A1F79E
+:100590003FE6E32E30E0F32E8AE890E026969FAF08
+:1005A0008EAF2697AE2DBF2D4D91EA2EFB2E042D3A
+:1005B000152D282D392D662D772D822D932DFCDEBE
+:1005C0002696EEADFFAD2697EE15FF0559F724965A
+:1005D0001FAE2497042D152D282D392D24964FADAF
+:1005E0002497662D772D8C2F9D2F0196E5DE282DE3
+:1005F000392D24964FAD2497662D772D8C2F9D2F66
+:100600000196DADEA62DB72D0D90BC91A02D28966F
+:10061000BFAFAEAF289723E0AA0FBB1F2A95E1F723
+:10062000E1E0F0E0EC0FFD1FAE0FBF1F11963C9113
+:10063000119712964C91129713965C9113971496FA
+:100640006C91149715967C911597FC912496FFAFA9
+:1006500024972696EEADFFAD2697A1902696FFAF84
+:10066000EEAF269724962FAD249716968C91169769
+:1006700017969C91B12CC12CD12CE12CF12C00E0CF
+:1006800010E04ED431F080918F00816080938F0014
+:1006900002C010928F002896EEADFFAD28973196DC
+:1006A000E770FF27A62DB72DED93FC932696EEADB0
+:1006B000FFAD2697EE58F04009F08CCF90E4E92E7C
+:1006C000F12C4114510421F480917000813318F40D
+:1006D00040918E0001C04DE0042D152D282D392D9F
+:1006E000662D772D8C2F9D2F019666DE81E0E81A0E
+:1006F000F10839F7E1E6F0E0A22DB32D8C91B8E0D6
+:100700002B0E311C819326143704B1F707C0812CBE
+:10071000912C2DCF82E3882E912C29CFC95BDF4FFE
+:100720000FB6F894DEBF0FBECDBFDF91CF911F9102
+:100730000F91FF90EF90DF90CF90BF90AF909F9080
+:100740008F907F906F905F904F903F902F90089583
+:100750002F923F924F925F926F927F928F929F92D1
+:10076000AF92BF92CF92DF92EF92FF920F931F93BF
+:10077000CF93DF93CDB7DEB729970FB6F894DEBFDE
+:100780000FBECDBF40917A0050917B0060917C00FC
+:1007900070917D0080917E0090917F00A0918000FB
+:1007A000B09181004090820050908300609084005E
+:1007B0007090850049825A826B827C82809086008C
+:1007C00090908700A0908800B090890050908F0092
+:1007D0005986C12CD12CE12CF12C212C312C442E0A
+:1007E000552E662E772E24E0440C551C661C771C73
+:1007F0002A95D1F7042F152F262F372FE5E03695B0
+:10080000279517950795EA95D1F740265126622638
+:100810007326440E551E661E771E3F2D2E2D1D2D50
+:100820000C2D0370112722273327000F111F000FF3
+:10083000111FE02FF12FEE58FE4F00811181228110
+:1008400033810C0D1D1D2E1D3F1D0425152526254C
+:100850003725800F911FA21FB31F07E4C01A06E8B7
+:10086000D00A08ECE00A01E6F00A3F2D2E2D1D2DDE
+:100870000C2DEBE03695279517950795EA95D1F75E
+:100880000370112722273327202F312F220F331FE8
+:10089000220F331FE22FF32FEE58FE4F00811181FC
+:1008A000228133814C2C5D2C6E2C7F2C400E511EEE
+:1008B000621E731E4D825E826F827886482E592E8C
+:1008C0006A2E7B2E14E0440C551C661C771C1A956E
+:1008D000D1F7082F192F2A2F3B2FE5E036952795C2
+:1008E00017950795EA95D1F704251525262537256F
+:1008F000080F191F2A1F3B1F4D805E806F80788470
+:100900000425152526253725400F511F621F731F0B
+:1009100007E020163104E1F488169906AA06BB0602
+:1009200051F449805A806B807C80441655066606D7
+:10093000770669F004C0882E992EAA2EBB2E81E07E
+:1009400080938F00BB2DAA2D992D882D0CC0198660
+:100950000FEF201A300A10E22116310409F03FCFC0
+:10096000298520938F00409361005093620060932B
+:100970006300709364008093650090936600A09379
+:100980006700B093680029960FB6F894DEBF0FBEDB
+:10099000CDBFDF91CF911F910F91FF90EF90DF902E
+:1009A000CF90BF90AF909F908F907F906F905F900F
+:1009B0004F903F902F900895CF92DF92EF92FF9249
+:1009C0000F931F93CF93C82ED92E00E010E0E62E90
+:1009D000F12C0E151F05E4F4EC2DFD2DE00FF11F99
+:1009E000C895802DFFDBC82FF1DB2C2F30E088234A
+:1009F00019F0822F932F03C0822F932F916004D17F
+:100A000084EC90E00197F1F700000F5F1F4FE1CFFA
+:100A1000CF911F910F91FF90EF90DF90CF900895AD
+:100A2000CF92DF92EF92FF920F931F93CF93DF93BA
+:100A3000C82ED92EE82EF92E062F10E08E2D9F2DD0
+:100A40008C199D0980179107C4F4EE2DFF2D81911B
+:100A5000EE2EFF2EC7DBC82FD0E0B8DB811104C01B
+:100A60008C2F9D2F916002C08C2F9D2FCDD084ECB8
+:100A700090E00197F1F70000E1CFDF91CF911F9156
+:100A80000F91FF90EF90DF90CF900895CF93DF9379
+:100A900080917000843709F46EC050F48823B1F05F
+:100AA000813009F0A0C0609173008FE690E016C01D
+:100AB0008A37B1F08C3709F458C0883709F093C0F1
+:100AC000609173006F5F80E690E008C081E09091D4
+:100AD0006C00911180E080936C0085C0A1DF83C021
+:100AE00080918F0060917300882321F06F5F80EA0E
+:100AF00090E039C080916B0081110BC080916E0035
+:100B0000853070F58F5F80936E006F5F8CE391E0AE
+:100B10002AC0813059F480916E00853008F58F5FCE
+:100B200080936E006F5F82E291E01DC0823059F4C5
+:100B300080916E008530A0F48F5F80936E006F5FB0
+:100B400088E091E010C0833059F480916E008530C8
+:100B500038F48F5F80936E006F5F8EEE90E003C07D
+:100B60006F5F8AEB90E028DF3EC0609173006F5F9B
+:100B700081E791E0F8CF80916C00882311F081ED3E
+:100B800001C084E790E091603FD0CFE6D0E031D063
+:100B900029DB899380E0CF38D807C9F780916F00AF
+:100BA000883F21F484E080936B001DC09091750014
+:100BB0009C30A9F018F4992351F015C0903239F007
+:100BC000933689F482E080936B00C2DD0CC081E033
+:100BD000911180E080936B0090E080DC04C083E0A2
+:100BE00080936B006BDB62E08EE990E0DF91CF9148
+:100BF000E3CE80916F018823E1F3809170019091A1
+:100C0000710110926F01089520916C012130E1F380
+:100C100090936E0180936D0181E080936C01089543
+:100C20001FBC89E08EBD8BE790E09BBD8ABD80E450
+:100C300089BFC698BE9810926F0110926C010895FA
+:100C40001F920F920FB60F9211242F933F934F9341
+:100C50005F938F939F93AF93BF93EF93FF932091F5
+:100C60006B01822F90E027FF3AC0E82FF92FE05860
+:100C7000F109E232F10540F5E35FFF4F099480E8A6
+:100C800080936B0180916C01811103C010926B0104
+:100C900068C080916D0190916E01880F991F9460DA
+:100CA00090936A018093690110926C0180916901AF
+:100CB00090916A01282F217020936801969587955D
+:100CC00090936A018093690180916801882311F0F3
+:100CD000BE9801C0BE9A80916B018F5F19C036B378
+:100CE00030743093680180329105C0F5E82FF92FF8
+:100CF000E15DFF4F099480916C01813011F480E82F
+:100D000007C0B6992EC010926A011092690181E065
+:100D100080936B0126C080916F018111B7CF8091C4
+:100D2000690190916A01909371018093700181E053
+:100D300080936F01ABCF8091690190916A01969584
+:100D40008795311103C040E050E002C040E051E01F
+:100D5000842B952B90936A01809369012F5F2093D8
+:100D60006B01FF91EF91BF91AF919F918F915F9137
+:100D70004F913F912F910F900FBE0F901F9018959C
+:100D80004FDF789460E182E891E016DECFE6D0E0B4
+:100D900060E2B62E76E0A72EEBE09E2E02E811E090
+:100DA000FAE18F2E80916F00833529F022DF1ADA65
+:100DB00080936F00F7CF50E7E52E50E0F52E19DF56
+:100DC00011DAEE2DFF2D8193EE2EFF2EF4E7EF16B4
+:100DD000F0E0FF06A1F75ADE8091690087FD37C079
+:100DE00090916B009130E9F078F0933059F5082E2E
+:100DF000000C990B23E0880F991F2A95E1F7682FC3
+:100E0000792F68567F4F19C0082E000C990B43E0CC
+:100E1000880F991F4A95E1F7682F792F685C7F4FFB
+:100E20000CC0082E000C990B33E0880F991F3A95DF
+:100E3000E1F7682F792F6050704048E050E080570C
+:100E40009F4F77D080916900815080936900809195
+:100E50006B0090916A00981769F180936A00109274
+:100E60006E00813059F038F0823059F0833011F53E
+:100E7000809269001FC0909269001CC0A092690016
+:100E800019C080E7C82E81E0D82E92E7E92E91E0C4
+:100E9000F92E8C2D9D2D60D0EE2DFF2D6193719339
+:100EA00081939193EE2EFF2EF4E0CF0ED11C0E1500
+:100EB0001F0579F7EC2FFD2F8B2D11928A95E9F7FD
+:100EC00071CF0024552704C0080E591F880F991FA1
+:100ED000009729F076956795B8F37105B9F7802DDD
+:100EE000952F089597FB10F8169400080F9308307B
+:100EF00098F00850232F342F452F562F672F782F27
+:100F0000892F902DF4CF05949795879577956795C0
+:100F100057954795379527950A95AAF70F91089504
+:100F20002A153B054C055D056E057F058007910779
+:100F30000895A82FB92F862F972FE82FF92FE19921
+:100F4000FECF06C0FFBBEEBBE09A31960DB20D920C
+:100F500041505040B8F70895A6E1B0E044E050E0B9
+:060F6000ECCFF894FFCF76
+:0E0F6600780000000000000000FF09090100F3
+:00000001FF
diff '-x=.git' -uNr hacktv/keyboard.c hacktv_patched/keyboard.c
--- hacktv/keyboard.c	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/keyboard.c	2020-07-13 13:21:25.992166700 +0200
@@ -0,0 +1,22 @@
+#include "keyboard.h"
+
+void _keyboard_raw_mode()
+{
+    tcgetattr(0, &term);
+    term.c_lflag &= ~(ICANON | ECHO); // Disable echo as well
+    tcsetattr(0, TCSANOW, &term);
+}
+
+void _keyboard_dis_raw_mode()
+{
+    tcgetattr(0, &term);
+    term.c_lflag |= ICANON | ECHO;
+    tcsetattr(0, TCSANOW, &term);
+}
+
+_Bool _kbhit()
+{
+    int byteswaiting;
+    ioctl(0, FIONREAD, &byteswaiting);
+    return byteswaiting > 0;
+}
diff '-x=.git' -uNr hacktv/keyboard.h hacktv_patched/keyboard.h
--- hacktv/keyboard.h	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/keyboard.h	2020-07-13 13:21:25.995160000 +0200
@@ -0,0 +1,15 @@
+/*KEYBOARD*/
+
+#ifndef KEYB_H
+#define KEYB_H
+
+#include <termios.h>
+#include <sys/ioctl.h>
+
+struct termios term;
+
+extern void _keyboard_raw_mode();
+extern void _keyboard_dis_raw_mode();
+extern _Bool _kbhit();
+
+#endif
diff '-x=.git' -uNr hacktv/play.sh hacktv_patched/play.sh
--- hacktv/play.sh	1970-01-01 01:00:00.000000000 +0100
+++ hacktv_patched/play.sh	2020-07-13 13:51:17.751779600 +0200
@@ -0,0 +1 @@
+./hacktv -o - test -m pal -t uint8 --wss auto --syster premiere-ca --showecm | vlc --demux rawvideo --rawvid-fps 25 --rawvid-width 1024 --rawvid-height 625 --rawvid-chroma=GREY --rawvid-aspect-ratio 1:1 -
diff '-x=.git' -uNr hacktv/syster.c hacktv_patched/syster.c
--- hacktv/syster.c	2020-07-13 13:56:09.767931100 +0200
+++ hacktv_patched/syster.c	2020-07-13 13:34:29.962161900 +0200
@@ -81,6 +81,19 @@
  *
 */
 
+
+
+/* keyboard stuff */
+#include "keyboard.h"
+_Bool encryption;
+_Bool cutrotate;
+_Bool debug;
+char taste;
+uint8_t crmode;
+uint8_t dnr = 2;
+int off = 0;
+
+
 /* 0 - 12.8 kHz complex FIR filter taps, sample rate 32 kHz */
 
 #define NTAPS 771
@@ -402,6 +415,29 @@
 void _render_ng_vbi(ng_t *s, int line, int mode)
 {
 	int x;
+	int offset = 0;
+
+	_keyboard_raw_mode(); /* Keyboard raw mode */
+	if(_kbhit()){
+		taste = getchar();
+		if(taste == ' '){
+			fprintf(stderr," ENCRYPTION-MODE CHANGE %d\n",!encryption);
+			encryption = !encryption;
+		}else if(taste == 'd'){
+			fprintf(stderr," DEBUG-MODE CHANGE %d\n",!debug);
+			debug = !debug;
+		}else if(taste == 'c'){
+			fprintf(stderr," CUT&ROTATE-MODE CHANGE %d\n",!cutrotate);
+			cutrotate = !cutrotate;
+		}else{}
+	} else {}
+
+	if(encryption){
+		crmode = mode;
+	} else {
+		crmode = 0x48;
+	}
+	_keyboard_dis_raw_mode();
 
 	/* Render the VBI data
 	 * These lines were used by Premiere */
@@ -428,7 +464,11 @@
 			}
 			
 			/* Build part 1 of the VBI block */
-			msg1[ 0] = mode | ((s->audience >> 5) & 1);	/* Decoder parameters + audience */
+                        if(cutrotate){
+                              msg1[ 0] = 0x74 | ((s->audience >> 5) & 1); // msg1[ 0] = 0x74;        /* Mode? (0x72 = Premiere / Canal+ Old, 0x48 = Clear, 0x7A or 0xFA) */
+                        }else{
+                              msg1[ 0] = crmode | ((s->audience >> 5) & 1); // msg1[ 0] = crmode;
+                        }
 			_ecm_part(s, &msg1[1]);
 			msg1[ 1] |= s->audience << 3; /* Audience uses 5 top bits of msg1[1] */
 			msg1[11] = 0xFF;	/* Simple checksum -- the Premiere VBI sample only has 0x00/0xFF here */
@@ -440,7 +480,11 @@
 			
 			/* Build part 2 of the VBI block */
 			msg2[ 0] = 0xFE;	/* ??? Premiere DE: 0xFE, Canal+ PL: 0x00 */
-			msg2[ 1] = 0x28;	/* ??? Premiere DE: 0x28, Canal+ PL: 0x2A, cut and rotate: 0x29 */
+		if(cutrotate){
+			msg2[ 1] = 0x29;	/* ??? Premiere DE: 0x28, Canal+ PL: 0x2A, cut and rotate: 0x29 */
+		}else{
+			msg2[ 1] = 0x28;        /* ??? Premiere DE: 0x28, Canal+ PL: 0x2A, cut and rotate: 0x29 */
+		}
 			msg2[ 2] = 0xB1;	/* ??? Premiere DE: 0xB1, Canal+ PL: 0xE4 */
 			msg2[ 3] = emm1 == _ppua_emm ? 0x01 : 0x00;	/* 0x00, or 0x01 when a broadcast EMM is present */
 			msg2[ 4] = emm2 == _ppua_emm ? 0x01 : 0x00;
@@ -459,6 +503,12 @@
 			/* Advance the block sequence counter */
 			s->block_seq++;
 		}
+if(debug){
+		fprintf(stderr,"\n");
+		for(int i =0;i<28;i++){fprintf(stderr,"%02X ",s->vbi[s->vbi_seq][i]);}
+		fprintf(stderr,"   S: %d R: %d VBI-LINE: %d FRAME: %d",s->s,s->r,line,s->vid->frame %25);
+}
+
 		
 		/* Render the line */
 		vbidata_render_nrz(s->lut, s->vbi[s->vbi_seq++], -45, NG_VBI_BYTES * 8, VBIDATA_LSB_FIRST, s->vid->output, 2);
@@ -699,6 +749,8 @@
 	f = (line < D11_FIELD_2_START ? 0 : 1);
 	i = line - (f == 0 ? D11_FIELD_1_START : D11_FIELD_2_START);
 	d11_field = (s->vid->frame % 3) + (s->vid->frame % 3) + f;
+
+	if(!encryption)i=0;
 	
 	if(i > 0 && i < D11_LINES_PER_FIELD)
 	{
@@ -927,6 +979,8 @@
 	}
 	
 	vid_adj_delay(s->vid, NG_DELAY_LINES);
+
+	if(!encryption || cutrotate){j=0;}
 	
 	/* Swap the active line with the oldest line in the delay buffer,
 	 * with active video offset in j if necessary. */
diff '-x=.git' -uNr hacktv/videocrypt.c hacktv_patched/videocrypt.c
--- hacktv/videocrypt.c	2020-07-13 13:56:09.778901000 +0200
+++ hacktv_patched/videocrypt.c	2020-07-13 13:49:01.501189900 +0200
@@ -46,6 +46,9 @@
 #include <math.h>
 #include "video.h"
 
+uint8_t keyctr;
+uint8_t eeprom;
+
 /* Packet header sequences */
 static const uint8_t _sequence[8] = {
 	0x87,0x96,0xA5,0xB4,0xC3,0xD2,0xE1,0x87,
@@ -92,7 +95,9 @@
 		0x07, 0,
 		{
  			{ 0x20 }, 
-			 { }, { }, { }, { },{ },
+			{ }, 
+			{ 0xC0,0xF1,0x00,0x44,0x33,0x22,0x11,0x00,0xFF,0xFF,0xFF }, 
+			{ }, { }, { },
  			{ 0xe8,0x43,0x0a,0x88,0x82,0x61,0x0c,0x29,0xe4,0x03,0xf6 }
 		},
 	},
@@ -100,7 +105,9 @@
 		0x07, 0,
 		{
  			{ 0x20,0x00,0x77,0x20,0x20,0x20,0x48,0x41,0x43,0x4b,0x54,0x56,0x20,0x20,0x20,0x20,0x53,0x4b,0x59,0x30,0x39,0x20,0x4d,0x4f,0x44,0x45 },
- 			{ }, { }, { }, { },	{ },
+ 			{ },
+			{ 0xC0,0xF1,0x00,0x44,0x33,0x22,0x11,0x00,0xFF,0xFF,0xFF }, 
+			{ }, { }, { },
  			{ 0xe8,0x43,0x0a,0x88,0x82,0x61,0x0c,0x29,0xe4,0x03,0xf6 }
 		},
 	},
@@ -231,7 +238,9 @@
 		0x07, 0,
 		{
 		 	{ 0x20 },
- 			{ }, { }, { }, { },	{ },
+ 			{ }, 
+			{ 0xC0,0xF1,0x00,0x44,0x33,0x22,0x11,0x00,0xFF,0xFF,0xFF },  
+			{ }, { }, { },
  			{ 0xE8,0x49,0x09,0x92,0x1F,0x93,0x20,0x74,0xCE,0x62,0xE4 }
 		}
 	},
@@ -239,7 +248,9 @@
 		0x07, 0,
 		{
 			{ 0x20,0x00,0x76,0x20,0x20,0x20,0x48,0x41,0x43,0x4b,0x54,0x56,0x20,0x20,0x20,0x20,0x20,0x54,0x41,0x43,0x20,0x4d,0x4f,0x44,0x45 },
- 			{ }, { }, { }, { },	{ },
+ 			{ }, 
+			{ 0xC0,0xF1,0x00,0x44,0x33,0x22,0x11,0x00,0xFF,0xFF,0xFF },  
+			{ }, { }, { },
  			{ 0xE8,0x49,0x09,0x92,0x1F,0x93,0x20,0x74,0xCE,0x62,0xE4 }
 		}
 	},
@@ -730,12 +741,40 @@
 			{
 				/* The active message is updated every 8th frame. The last
 				 * message in the block is a duplicate of the first. */
+				if(strcmp(mode,"tac1") == 0 || strcmp(mode,"tac2") == 0){
+					/* KEY-UPDATE as plaintext */
+					if((s->counter & 0x3F) == 0x10)
+					{ fprintf(stderr,"\nKEY-UPDATE BLOCK \n");
+						s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][2] = eeprom+1;
+						for(x = 0; x < 16; x++){ s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][11+x] = tac_key[keyctr] & 0xff; keyctr++;}
+						eeprom++;
+						if(keyctr == 96){keyctr= 0;eeprom = 0;}
+					}
+                                        /* TODO:ENCRYPT KEYUPDATE */
+				}
+
+				if(strcmp(mode,"sky09") == 0){
+                                        /* KEY UPDATE as plaintext */
+					if((s->counter & 0x3F) == 0x10)
+                                        { fprintf(stderr,"\nKEY-UPDATE BLOCK \n");
+                                                s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][2] = eeprom+7;
+                                                for(x = 0; x < 16; x++){ s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][11+x] = sky09_key[keyctr] & 0xff; keyctr++;}
+                                                eeprom++;
+						if(keyctr == 224){keyctr= 0;eeprom = 0;}
+                                        }
+					/* TODO:ENCRYPT KEYUPDATE */
+                }
+				
+				/* CRC */
 				for(crc = x = 0; x < 31; x++)
 				{
 					crc += s->message[x] = s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][x];
+					//fprintf(stderr,"%02x ",s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][x]);
 				}
 				
 				s->message[x] = ~crc + 1;
+				//fprintf(stderr,"%02x ",s->blocks[s->block].messages[((s->counter >> 3) & 7) % 7][31]);				
+				//fprintf(stderr,"Codeword: %08llx\n", s->cw);
 			}
 			
 			if((s->counter & 4) == 0)
@@ -767,9 +806,12 @@
 				for(crc = x = 0; x < 31; x++)
 				{
 					crc += s->message2[x] = s->blocks2[s->block2].messages[(s->counter >> 1) & 7][x];
+				//	fprintf(stderr,"%02x ",s->blocks2[s->block2].messages[(s->counter >> 1) & 7][x]);
 				}
 				
 				s->message2[x] = ~crc + 1;
+				//fprintf(stderr,"%02x ",s->blocks2[s->block2].messages[(s->counter >> 1) & 7][31]);				
+				//fprintf(stderr,"Codeword: %08llx\n", s->cw);
 			}
 			
 			if((s->counter & 1) == 0)
@@ -1382,4 +1424,4 @@
 	
 	/* Reverse calculated control word */
 	s->codeword = ((uint64_t) v0 << 32 | v1) & 0x0fffffffffffffffUL;
-}
\ No newline at end of file
+}
